packages/code-du-travail-frontend/src/common/RelatedItems.js

```diff
@@ -1,165 +1,97 @@
-                  />
-                </Link>
-              ) : (
-                <CallToActionTile
-                  action={action || "Consulter"}
-                  custom={false}
-                  href={url}
-                  icon={icons[icon]}
-                  rel="noopener noreferrer nofollow"
-                  subtitle={subtitle}
-                  target="_blank"
-                  title={title}
-                >
-                  {description}
-                </CallToActionTile>
-              )}
-            </StyledTileItem>
-          )
-        )}
-      </StyledFlatList>
-      {relatedLinkItems.length > 0 && (
-        <>
-          <Heading as="div">
-            Les articles pouvant vous intéresser&nbsp;:
-          </Heading>
-          <FlatList>
-            {relatedLinkItems.map(({ slug, source, title, reco }) => {
-              let rootSlug = slug;
-              let hash;
-              if (slug.includes("#")) {
-                [rootSlug, hash] = slug.split("#");
-              }
-              hash = hash ? `#${hash}` : "";
-              rootSlug = rootSlug ? `/${rootSlug}` : "";
-              const route = getRouteBySource(source);
+                    hash = hash ? `#${hash}` : "";
+                    rootSlug = rootSlug ? `/${rootSlug}` : "";
+                    const route = getRouteBySource(source);
+
+                    href = `/${route}${rootSlug ? `/${slug}` : ""}`;
+                    selection = `${route}${rootSlug}${hash}`;
+                  } else {
+                    href = url;
+                    selection = url;
+                  }
 
-              return (
-                <StyledLinkItem key={slug}>
-                  <Link
-                    href={`/${route}${rootSlug ? "/[slug]" : ""}`}
-                    as={`/${route}${rootSlug}${hash}`}
-                    passHref
-                  >
-                    <ArrowLink
-                      rel="nofollow"
-                      arrowPosition="left"
-                      onClick={() =>
-                        matopush([
-                          "trackEvent",
-                          "selectRelated",
-                          JSON.stringify({
-                            reco,
-                            selection: `${route}${rootSlug}${hash}`,
-                          }),
-                        ])
-                      }
-                    >
-                      {title}
-                    </ArrowLink>
-                  </Link>
-                </StyledLinkItem>
-              );
-            })}
-          </FlatList>
-        </>
+                  return (
+                    <StyledLinkItem key={slug || url}>
+                      <Link href={href} passHref>
+                        <ArrowLink
+                          rel="nofollow"
+                          arrowPosition="left"
+                          onClick={() => matoSelectRelated(reco, selection)}
```

Comment:
maybe that selection is in that href and don't need to be in a separate variable

Reply:
So, that's a bit more tricky, in fact `href` was the variable without the `#section` in the next Link `href/as`.
What we miss here is the `as`, I will restore it.

Reply:
Before putting back the "as", are you sure that adding the "#" directly in href will break routing ? I wouldn't be surprised if it worked